dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/vitals/MemoryVitalReaderTest.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.vitals
+
+import com.datadog.android.utils.forge.Configurator
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.io.File
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.junit.jupiter.api.io.TempDir
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class MemoryVitalReaderTest {
+
+    lateinit var testedReader: VitalReader
+
+    @TempDir
+    lateinit var tempDir: File
+
+    lateinit var fakeFile: File
+
+    @StringForgery(regex = "(\\.[a-z]+)+")
+    lateinit var fakeName: String
+
+    @StringForgery(regex = "[RSDZTtWXxKWP]")
+    lateinit var fakeState: String
+
+    @IntForgery(1)
+    var fakePid: Int = 0
+
+    @IntForgery(1, 0x7F)
+    var fakeVmRss: Int = 0
+
+    @IntForgery(1, 256)
+    var fakeThreads: Int = 0
+
+    lateinit var fakeStatusContent: String
+
+    @BeforeEach
+    fun `set up`() {
+        fakeFile = File(tempDir, "stat")
+        fakeStatusContent = generateStatusContent()
+        testedReader = MemoryVitalReader(fakeFile)
+    }
+
+    @Test
+    fun `𝕄 read correct data 𝕎 readVitalData()`() {
+        // Given
+        fakeFile.writeText(fakeStatusContent)
+
+        // When
+        val result = testedReader.readVitalData()
+
+        // Then
+        Assertions.assertThat(result).isEqualTo(fakeVmRss.toDouble())
+    }
+
+    @Test
+    fun `𝕄 read correct data 𝕎 readVitalData() {multiple times}`(
+        @IntForgery(1) vmRssValues: List<Int>
+    ) {
+        // Given
+        val results = mutableListOf<Double>()
+        // When
+        vmRssValues.forEach { vmRss ->
+            fakeVmRss = vmRss
```

Comment:
looks weird to update a local variable to generate those fake data. Why not passing this value as a parameter to `generateStatusContent` function ?